endpoint.go

```diff
@@ -46,13 +49,29 @@ type EndpointOption func(ep *endpoint)
 // ContainerData is a set of data returned when a container joins an endpoint.
 type ContainerData struct {
 	SandboxKey string
-	HostsPath  string
 }
 
 type containerConfig struct {
-	hostName   string
-	domainName string
-	generic    map[string]interface{}
+	hostName       string
+	domainName     string
+	generic        map[string]interface{}
+	hostsPath      string
+	ExtraHosts     []extraHost
+	parentUpdates  []parentUpdate
+	resolvConfPath string
+	dnsList        []string
+	dnsSearchList  []string
```

Comment:
These groups of data are useful for managing the container's /etc/hosts and resolve.conf. I think it will be useful to group them and add comments on why they are required. makes the code more readable I guess :)